syntax = "proto3";

message ImageRequest {
  string description = 1;  // User-provided text (e.g., "small white dog")
}

message ImageResponse {
  bytes image_data = 1;  // Binary data of the image
  string image_url = 2;   // Add this line for the image URL
}

service ImageSearch {
  rpc SearchImage (ImageRequest) returns (ImageResponse);
}
